tests/benchmarks/benchmark_runner.py

```diff
@@ -154,43 +156,66 @@ async def test_benchmark(retries, benchmarks):
         start_commit = repo.commit()
         repo.git.checkout(benchmark.commit)
 
-        for prompt in benchmark.prompts:
+        for i, prompt in enumerate(benchmark.prompts):
             print("  Prompt:", prompt)
-            results[title][prompt] = {
-                "Verified": 0,
-            }
-            for i in range(1, retries + 1):
+            for j in range(1, retries + 1):
+                formatted_title = re.sub(r"[ '\"/\\-^]", "", title).replace(" ", "_")
+                result = BenchmarkResult(
+                    name=f"{formatted_title}-{i}-{j}",
+                    family=formatted_title,
+                )
                 client = PythonClient(cwd=codebase, config=benchmark.config)
                 await client.startup()
                 response = await client.call_mentat(prompt)
                 await client.call_mentat("y")
                 await client.wait_for_edit_completion()
 
                 await client.call_mentat("q")
+                messages = client.get_conversation().literal_messages
+                cost_tracker = client.get_cost_tracker()
+                result.cost = cost_tracker.total_cost
+                result.tokens = cost_tracker.total_tokens
+                result.transcript = {
+                    "id": result.name,
+                    "messages": messages,
+                }
+
                 await client.shutdown()
 
                 if hasattr(benchmark, "verify"):
-                    if benchmark.verify():
-                        results[title][prompt]["Verified"] += 1
+                    result.verify = benchmark.verify()
 
-                diff = repo.git.diff(start_commit)
+                # Set syntax and response grade information
+                diff = repo.git.diff()
```

Comment:
afaik this only saves edits, not new/untracked files. I ran into the same thing in #374 and added a helper func in git_handler. (I think this is why the code was missing in my earlier comment)

Reply:
Now i see you already caught it. I think stage/diff/unstage is the right approach too.